extension/app/background.ts

```diff
@@ -1,3 +1,114 @@
+import { randomString } from "./src/lib/utils";
+import {
+  AUTH0_AUDIENCE,
+  AUTH0_CLIENT_DOMAIN,
+  AUTH0_CLIENT_ID,
+  AuthBackroundMessage,
+  Auth0AuthorizeResponse,
+  AuthBackgroundResponse,
+} from "./src/lib/auth";
+
+/**
+ * Listener to open/close the side panel when the user clicks on the extension icon.
+ */
 chrome.runtime.onInstalled.addListener(() => {
   void chrome.sidePanel.setPanelBehavior({ openPanelOnActionClick: true });
 });
+
+/**
+ * Listener for messages sent from the react app to the background script.
+ * For now we use messages to authenticate the user.
+ */
+chrome.runtime.onMessage.addListener(
+  (
+    message: AuthBackroundMessage,
+    sender,
+    sendResponse: (
+      response: Auth0AuthorizeResponse | AuthBackgroundResponse
+    ) => void
+  ) => {
+    switch (message.type) {
+      case "AUTHENTICATE":
+        authenticate(sendResponse);
+        return true; // Keep the message channel open for async response.
+
+      case "LOGOUT":
+        logout(sendResponse);
+        return true; // Keep the message channel open.
+
+      default:
+        console.error(`Unknown message type: ${message.type}.`);
+    }
+  }
+);
+
+/**
+ * Authenticate the user using Auth0.
+ */
+const authenticate = async (
+  sendResponse: (auth: Auth0AuthorizeResponse | AuthBackgroundResponse) => void
+) => {
+  if (!AUTH0_CLIENT_ID || !AUTH0_CLIENT_DOMAIN) {
+    console.error("Auth0 client ID or domain is missing.");
+    return;
+  }
+
+  const redirectUrl = chrome.identity.getRedirectURL();
+  const options = {
+    client_id: AUTH0_CLIENT_ID,
+    response_type: "id_token token code",
+    scope: "offline_access openid profile email",
+    redirect_uri: redirectUrl,
+    nonce: randomString(16),
+    audience: AUTH0_AUDIENCE,
```

Comment:
At some point we must add a code_challenge attribute to support pkce (probably need to setup the flow type  in auth0 )

Reply:
+1 to this, as we expose the client ID and domain, PKCE might be a good solution here.

Reply:
Done! 👍🏻